contracts/referrals/ReferralGateway.sol

```diff
@@ -192,6 +200,15 @@ contract ReferralGateway is
         usdc = IERC20(_usdcAddress);
     }
 
+    function initializeNewVersion(
```

Comment:
It is not protected because usually the initializer are called in deployment scripts and reinitializer are called in upgrades calls. If it is protected then it should be called right after the upgrade